river/command/reset_config.zig

```diff
@@ -0,0 +1,66 @@
+// This file is part of river, a dynamic tiling wayland compositor.
+//
+// Copyright 2021 The River Developers
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+const util = @import("../util.zig");
+
+const server = &@import("../main.zig").server;
+
+const Seat = @import("../Seat.zig");
+const Config = @import("../Config.zig");
+
+const Error = @import("../command.zig").Error;
+
+pub fn resetConfig(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const [:0]const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len > 1) return Error.TooManyArguments;
+
+    // Create new config. If unsuccessful, the current config and state remain as is.
+    var new_config = try Config.init();
+    server.config.deinit();
+    server.config = new_config;
+
+    // Return all seats to normal mode.
+    {
+        var it = server.input_manager.seats.first;
+        while (it) |node| : (it = node.next) {
+            node.data.mode_id = 0;
+            node.data.prev_mode_id = 0;
```

Comment:
What if we are currently in locked mode due to an active input inhibitor?

Reply:
Ah, I forgot that.